src/main/java/com/develhope/spring/users/services/UserService.java

```diff
@@ -2,23 +2,43 @@
 
 import com.develhope.spring.users.components.UserMapper;
 import com.develhope.spring.users.dtos.UserCreatorDTO;
-import com.develhope.spring.users.dtos.UserSavedDTO;
+import com.develhope.spring.users.dtos.UserResponseDTO;
+import com.develhope.spring.users.dtos.UserReworkedDTO;
+import com.develhope.spring.users.dtos.UserUpdaterDTO;
+import com.develhope.spring.users.models.Roles;
 import com.develhope.spring.users.models.User;
 import com.develhope.spring.users.repositories.UserRepository;
+import com.develhope.spring.vehicles.responseStatus.NotAuthorizedOperationException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
 public class UserService {
     @Autowired
     private UserRepository userRepository;
     @Autowired
     private UserMapper userMapper;
 
+    private void checkUserAuthorizationBy(long userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        if (optionalUser.isEmpty() || !optionalUser.get().getRoles().equals(Roles.ADMIN)) {
+            throw new NotAuthorizedOperationException("Permesso negato. Non autorizzato ad aggiornare gli utenti");
+        }
+    }
 
-    public UserSavedDTO createUser (UserCreatorDTO userCreatorDTO) {
-        User userToRegister = userMapper.toEntityFrom(userCreatorDTO);
+    public UserResponseDTO createUser (UserCreatorDTO userCreatorDTO) {
+        User userToRegister = userMapper.toEntity(userCreatorDTO);
         userRepository.save(userToRegister);
-        return userMapper.toUserSavedDTOFrom(userToRegister);
+        return userMapper.toResponseDTO(userToRegister);
+    }
+
+    public UserReworkedDTO update(long userId, UserUpdaterDTO userUpdaterDTO) {
```

Comment:
eh scusa come diceva Lorenzo in questo caso per non passare l'id, ma per evitare anche di mettere l'id nel UpdaterDto, passa un veicolo con solo id nel costruttore che già cè e prendi l'id da quello, oppure crea un dto di veicolo con solo l'id.

Reply:
anche se, ora che devo fare una cosa simile, sorge il problema che non si possono passare due request body mi pare. Quindi sono un po confuso ora